libraries/chain/controller.cpp

```diff
@@ -3878,11 +3878,10 @@ struct controller_impl {
           });
    }
 
-   // Verify QC claim made by finality_extension in header extension
-   // and quorum_certificate_extension in block extension are valid.
+   // Verify basic proper_block invariants.
    // Called from net-threads. It is thread safe as signed_block is never modified after creation.
    // -----------------------------------------------------------------------------
-   void verify_proper_block_exts( const block_id_type& id, const signed_block_ptr& b, const block_header_state& prev ) {
+   void verify_basic_proper_block_invariants( const signed_block_ptr& b, const block_header_state& prev ) {
```

Comment:
I think this should return a `std::optional<qc_t>`. Then if the optional contains a `qc_t`, pass it into `verify_qc()`. Note `qc_t` has a method `to_qc_claim()` which is very cheap to call. That will be much faster and eliminate duplicate code.

Reply:
Thanks. I was trying to make the function independent. But your method is shorter and faster.